content/articles/hyperparameter-tuning-of-machine-learning-model-in-python/index.md

```diff
@@ -0,0 +1,347 @@
+<!--more-->
+This tutorial will increase the model's accuracy score. This ensures that the model makes accurate predictions. We will also create a list of all the possible values for hyperparameters and iterate through the values, finding all the hyperparameters combinations. We then calculate and record the performance of each parameter. Finally, we use hyperparameters that will provide an optimal model.
+
+### Table of contents
+- [Prerequisites](#prerequisites)
+- [Hyperparemeter tuning techniques](#hyperparemeter-tuning-techniques)
+- [Generate synthetic dataset](#generate-synthetic-dataset)
+- [Examine the data dimension](#examine-the-data-dimension)
+- [Split our dataset](#split-our-dataset)
+- [Building a machine learning model using Random Forest](#building-a-machine-learning-model-using-random-forest)
+- [Model fitting](#model-fitting)
+- [Making prediction using the testing dataset](#making-prediction-using-the-testing-dataset)
+- [Accuracy score](#accuracy-score)
+- [Getting started with hyperparameter tuning](#getting-started-with-hyperparameter-tuning)
+- [Creating the grid](#creating-the-grid)
+- [The best parameters for the model](#the-best-parameters-for-the-model)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites
+To follow along comfortably, a reader is required:
+- To have [Python](https://www.python.org/) installed.
+- To know [Python programming](/engineering-education/python-projects-for-beginners/)
+- To know how to train a [machine learning model](/engineering-education/house-price-prediction/)
+- To know how to work with the [Scikit-learn library.](https://scikit-learn.org/stable/)
+- To know how to work with [Google Colab notebook.](https://research.google.com/colaboratory/)
+
+### Hyperparameter tuning techniques
+Choosing the optimal hyperparameters is important in building a successful machine learning model. 
+Hyperparameters have a great impact on the machine learning algorithms used. Manual searching for the best hyper-parameter is a tedious process. Therefore, we need techniques that simplify this work. 
+
+These techniques are as follows.
+
+#### Grid Search
+This is a brute force searching technique. In this technique, we create a list of all the combination values for hyperparameters.  We then iterate through all hyperparameters. Finally, it records the best performing hyperparameters used in model training. This is shown below.
+
+![Grid Search Technique](/engineering-education/hyperparameter-tuning-of-machine-learning-model-in-python/grid-search-technique.png)
+
+*[Image source: Medium](https://miro.medium.com/max/534/1*6mLLIGYLny9B6GAnVZcltw.png)*
+
+#### Random Search
+In this technique, we also create a list of all the combination values for hyperparameters. It's similar to grid search, but uses random search instead of using exhaustive search.
```

Comment:
```suggestion
We also create a list of all the combination values for hyperparameters in this technique. It's similar to grid search, but it uses random search instead of exhaustive search.
```